code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -176,15 +176,15 @@
 		//Fill machine with a bag!
 		if(istype(I, /obj/item/storage/bag))
 				var/obj/item/storage/bag/B = I
-				for (var/obj/item/reagent_containers/food/snacks/grown/G in B.contents)
+				for (var/obj/item/reagent_containers/food/G in B.contents)
```

Comment:
You fix it for a single use case of pills, but there's still a lot of stuff that can fit into grinder, but can't be moved from bags (honeycomb, minerals, weird plants such as nettles).
I suggest putting a more generic check instead. Something like:
```js
for(var/item in B.contents)
  if(!is_type_in_list(item , blend_items) && !is_type_in_list(item, juice_items))
    continue
  // do the item move logic
```
This doesn't handle dried items well, though, since they need the dry check.. Except it does, since all the dried items happen to be a subtype of /food/snacks/grown, so they happen to be handled already... Except if you transfer them via a bag, you don't need to dry them... But at least you are not introducing a new bug here... But I digress... (i couldn't figure out a way to substantially improve this code, so this is probably fine?)

Reply:
Probably a better thing to do yeah. Honestly looking at the grinder code makes me want to just refactor/clean up the whole thing, lots of ugly crap in here but that's out of the scope of the PR.

Reply:
> but that's out of the scope of the PR
I agree. Feel free to simply resolve this comment if that's not something you want to tackle at the moment.

Reply:
I was referring to a complete grinder refactor/clean up being out of the scope. Your changes are a good idea, and I will be implementing them. :+1: